app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEvent.kt

```diff
@@ -19,3 +20,31 @@ object AppointmentDateDecremented : ScheduleAppointmentEvent() {
 data class AppointmentCalendarDateSelected(val selectedDate: LocalDate) : ScheduleAppointmentEvent() {
   override val analyticsName = "Schedule Appointment:Appointment calendar date selected"
 }
+
+object ManuallySelectAppointmentDateClicked : ScheduleAppointmentEvent() {
+  override val analyticsName = "Schedule Appointment:Manually Select Appointment Date"
+}
+
+data class CurrentFacilityLoaded(val facility: Facility) : ScheduleAppointmentEvent()
+
+data class PatientFacilityChanged(val facility: Facility) : ScheduleAppointmentEvent() {
+  override val analyticsName = "Schedule Appointment: Patient facility changed"
+}
+
+object AppointmentScheduled : ScheduleAppointmentEvent()
+
+object AppointmentDone : ScheduleAppointmentEvent() {
+  override val analyticsName = "Schedule Appointment:Appointment done"
+}
+
+data class PatientDefaulterStatusLoaded(
+    // I don't really like the `isPatientADefaulter` name. What this
+    // flag actually means is: The patient is at high-risk AND they don't have a follow-up visit scheduled.
+
+    // What is a better name to indicate this?
```

Comment:
The [dictionary](https://www.google.com/search?client=firefox-b-d&q=defaulter) says that a defaulter is "a person who fails to complete a course of medical treatment".
This does explain it the way it is implemented. Also we use high-risk definition differently in different places so I'd much rather not include that here as well. Wdyt?

Reply:
Hmm, good points. I'll remove this comment then.

Reply:
Although

Reply:
Removed my comment for now, maybe we can resolve this discussion outside the scope of this PR and then take a call.

Reply:
Actually no. This is someone who hasn't shown up for a followup in 30 days.
The way we implement it is by scheduling an appointment for them in advance so as to remind them to come for check up.

Reply:
We can choose to follow the definition of what we have implemented rather than what it means as per the PRD. So one suggestion could be `shouldRemindForFollowUp`?

Reply:
Let's leave this for now then since everyone in the system uses `defaulter` for this.